etl/src/v2/destination/memory.rs

```diff
@@ -0,0 +1,28 @@
+use crate::conversions::cdc_event::CdcEvent;
+use crate::v2::destination::base::Destination;
+use std::sync::Arc;
+use tokio::sync::RwLock;
+
+#[derive(Debug)]
+struct Inner {
+    events: Vec<CdcEvent>,
+}
+
+#[derive(Debug, Clone)]
+pub struct MemoryDestination {
```

Comment:
We might want to move these as test utils in the future, unless some users of the library might benefit from it.